### ISSUE TITLE
Support for quadprog

### ISSUE THREAD

**Issue by doncwilliam** [cvxgrp/cvxpy#444]
Do you have plan to support quadprog? It seems its performance is very competitive against major QP solvers, and is open source.
The performance comparison is here:
https://github.com/stephane-caron/qpsolvers
Thank you!
Don

**Comment by rileyjmurray** [comment]
Can't speak to any plans the main developers might have, but from the link you provided, OSQP performs just about as well as quadprog (it even performs substantially better on sparse problems).

Are there other reasons for supporting quadprog? OSQP uses ADMM, so if quadprog uses a particularly robust interior-point solver then that may be a factor in its favor.

**Comment by sschnug** [comment]
Looks like a dense-only solver which is not an IPM. There are probably limited use-cases. (That being said, i would not see OSQP as the first-pick QP-solver; but alternatives are rare: cvxopt & ecos being the typical IPM-methods in a more general framework).

**Comment by rileyjmurray** [comment]
Gurobi, MOSEK and CPLEX fulfill the role of a robust interior-point solvers for QP's. The only drawback is that they're commercial solvers. Since speed seems comparable between quadprog and OSQP, the only thing that remains is robustness. Other free solvers like CVXOPT and ECOS are plenty robust, so it seems to be that quadprog doesn't fulfill a pressing need in CVXPY.

Of course, if quadprog is as fast as OSQP and as robust as CVXOPT / ECOS, then that changes things. @doncwilliam , can you point to any benchmarks or reports indicating that quadprog has this robustness?

**Comment by sschnug** [comment]
Some people don't have access to commercial solvers. OSQP is not as accurate or robust as an IPM. CVXOPT and ECOS are not as specialized as a pure CQP-solver could be (cvxopt's QP might be; but it's not as performant / robust as one would expect; probably one should check the theory/code).

**Comment by bstellato** [comment]
A bit biased reply here (OSQP developer). I believe that for many many practical applications OSQP is more than enough. That's why it will be the default one for CVXPY 1.0. I tried many QP solvers in the last months (see [some benchmarks here](https://github.com/oxfordcontrol/osqp_benchmarks)). 

Interior Point methods sometimes fail badly giving for numerical errors. This is because the linear system that solved at each iteration is badly scaled. ECOS is one example (see  https://github.com/embotech/ecos/issues/150 https://github.com/embotech/ecos/issues/142 https://github.com/embotech/ecos/issues/134). @sschung if for robustness you mean not having numerical issues, I can assure that first-order methods are very robust. However, they can take more iterations if the step-size parameters are not chosen correctly. 

That being said, I would be happy to have many QP solvers supported in CVXPY because it makes it easier to compare and improve them. That's why I wrote most of the `*_qpif.py` files in the 1.0 branch. However, I would support mostly sparse solvers since dense ones like qpOASES and quadprog tend to become really slow when the dimensions increase. @rileyjmurray @doncwilliam
